RadarSDK/RadarCoordinate.m

```diff
@@ -9,6 +9,63 @@
 
 @implementation RadarCoordinate
 
++ (NSArray<RadarCoordinate *> *_Nullable)coordinatesFromObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSArray class]]) {
+        return nil;
+    }
+
+    NSArray *coordinatesArr = (NSArray *)object;
+
+    NSMutableArray<RadarCoordinate *> *mutableCoordinates = [NSMutableArray<RadarCoordinate *> new];
+
+    for (id coordinateObj in coordinatesArr) {
+        RadarCoordinate *coordinate = [[RadarCoordinate alloc] initWithObject:coordinateObj];
+
+        if (!coordinate) {
+            return nil;
+        }
+
+        [mutableCoordinates addObject:coordinate];
+    }
+
+    return mutableCoordinates;
+}
+
+- (instancetype _Nullable)initWithObject:(id)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSDictionary *dict = (NSDictionary *)object;
+
+    id coordinatesObj = dict[@"coordinates"];
+    if (!coordinatesObj || ![coordinatesObj isKindOfClass:[NSArray class]]) {
+        return nil;
+    }
+
+    NSArray *coordinatesArr = (NSArray *)coordinatesObj;
+    if (coordinatesArr.count != 2) {
+        return nil;
+    }
+
+    id coordinatesLongitudeObj = coordinatesArr[0];
+    id coordinatesLatitudeObj = coordinatesArr[1];
+    if (!coordinatesLongitudeObj || !coordinatesLatitudeObj || ![coordinatesLongitudeObj isKindOfClass:[NSNumber class]] ||
+        ![coordinatesLatitudeObj isKindOfClass:[NSNumber class]]) {
+        return nil;
+    }
+
+    NSNumber *coordinatesLongitudeNumber = (NSNumber *)coordinatesLongitudeObj;
+    NSNumber *coordinatesLatitudeNumber = (NSNumber *)coordinatesLatitudeObj;
+
+    float coordinatesLongitudeFloat = [coordinatesLongitudeNumber floatValue];
+    float coordinatesLatitudeFloat = [coordinatesLatitudeNumber floatValue];
+
```

Comment:
nit: there is a helper method `[NSDictionary radar_coordinateForKey]` in `RadarCollectionsAddition`, which does a bit more thorough checking on coordinate fields from server response

Reply:
Good call, will use that here